src/main/java/hudson/plugins/testlink/AbstractTestLinkBuilder.java

```diff
@@ -361,4 +371,26 @@ public Action getProjectAction(AbstractProject<?, ?> project) {
         return customFieldNamesArray;
     }
 
+
+    protected String[] createArrayOfTestPlanCustomFieldsNames(final VariableResolver<String> variableResolver,
```

Comment:
Will try to merge createArrayOfTestPlanCustomFieldsNames and createArrayOfCustomFieldsNames methods.

You could have done that, but that would make this pull request harder to merge. By simply copying and pasting, you made it easier for me to review your code, and also to spot parts of the code that were not suitable for extensibility.

Kudos and thanks! :+1: